### PR TITLE
vcpkg (new formula)

### PR DESCRIPTION
**Pull request by flier**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
bump version to 2019.08 base on #36347

### CODE REVIEW

**Comment by equal-l2** [comment]
```
==> /usr/local/Cellar/vcpkg/2019.08/bin/vcpkg version
dyld: Symbol not found: __ZNKSt3_V214error_category10_M_messageB5cxx11Ei
  Referenced from: /usr/local/Cellar/vcpkg/2019.08/bin/vcpkg
  Expected in: /usr/lib/libstdc++.6.0.9.dylib
 in /usr/local/Cellar/vcpkg/2019.08/bin/vcpkg
Error: vcpkg: failed
```

Seems that something is wrong with compilation.

File: Formula/vcpkg.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    url "https://github.com/microsoft/vcpkg/archive/2019.08.tar.gz"
+    sha256 "d533d8b577ce527c018aaaa77cee88d1940e3772221b7a6b4b995d3e953730e1"
+  end
+
+  head do
+    url "https://github.com/microsoft/vcpkg.git"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "gcc" => :build
```
Comment (fxcoudert) [inline Formula/vcpkg.rb]: Does it not build with the system compiler (clang/LLVM)?

**Comment by flier** [comment]
It seems broken when compile with latest XCode update microsoft/vcpkg#7585 :(

File: Formula/vcpkg.rb (hunk 1 above)
Comment (flier) [reply Formula/vcpkg.rb]: It seems `vcpkg` 2019.08 used some C++ features that system clang/LLVM doesn't support yet.

```
./bootstrap-vcpkg.sh -useSystemBinaries -allowAppleClang
Last 15 lines from /Users/fl0120/Library/Logs/Homebrew/vcpkg/01.bootstrap-vcpkg.sh:
In file included from ../src/vcpkg/archives.cpp:3:
In file included from ../include/vcpkg/archives.h:3:
../include/vcpkg/base/files.h:96:29: error: 'path' is unavailable: introduced in macOS 10.15
void is_directory(const fs::path& p, std::error_code& ec) = delete;
                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/filesystem:739:24: note: 'path' has been explicitly marked unavailable here
class _LIBCPP_TYPE_VIS path {
                       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

**Comment by fxcoudert** [comment]
> It seems broken when compile with latest XCode update

We're likely to wait until it compiles with latest Xcode.

**Comment by flier** [comment]
> > It seems broken when compile with latest XCode update
> 
> We're likely to wait until it compiles with latest Xcode.

I guess their will release 2018.9 at end of month, let's wait a few days

**Comment by SMillerDev** [comment]
Any update?

**Comment by fxcoudert** [comment]
Closing for now.

**Pull request closed without merging**